### PR TITLE
Cleanup Documentation

### PR DESCRIPTION
**Pull request by hisaac**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

This change just updates the documentation in the `step.yml` file to better match our determined style.

### Changes

- Added `.vscode` to the `.gitignore`
- Slimmed down the step's description to only include mention of commonly modified inputs, rather than descriptions of every input.
- Updated some input titles and descriptions.

### Investigation details

I met with a member of the tech writers team to learn more about how best to document steps, and will have them review this PR prior to opening it.

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -70,42 +61,39 @@ toolkit:
...
+      title: Project (or Workspace) Path
+      summary: The path to your project's `.xcodeproj` or `.xcworkspace`
+      description: The path to your project's `.xcodeproj` or `.xcworkspace`
       is_required: "true"
   - scheme: $BITRISE_SCHEME
     opts:
-      title: Scheme name
-      summary: "The Scheme to use."
-      description: "The Scheme to use."
+      title: Scheme Name
```
Comment (zoltan-baba) [inline step.yml:70]: name

File: step.yml [hunk 2]
```diff
@@ -70,42 +61,39 @@ toolkit:
...
+      summary: The name of the Xcode Scheme to use when building your project
+      description: The name of the Xcode Scheme to use when building your project
       is_required: "true"
   - simulator_device: iPhone 8 Plus
     opts:
-      title: "Device model"
+      title: Device Model
```
Comment (zoltan-baba) [inline step.yml:76]: model

File: step.yml [hunk 3]
```diff
@@ -70,42 +61,39 @@ toolkit:
...
-        * Apple TV 4K (don't forget to set the platform to `tvOS Simulator` to use this option!)
+        * Apple TV 4K
+        
+        Don't forget to set the platform to `tvOS Simulator` in order to use an Apple TV simulator.
       is_required: "true"
-  - simulator_os_version: "latest"
+  - simulator_os_version: latest
     opts:
-      title: OS version
+      title: OS Version
```
Comment (zoltan-baba) [inline step.yml:92]: version

File: step.yml [hunk 4]
```diff
@@ -128,33 +114,19 @@ inputs:
       is_required: "true"
   - configuration:
     opts:
-      title: "Configuration name"
+      title: "Configuration Name"
```
Comment (zoltan-baba) [inline step.yml:117]: name

File: step.yml [hunk 5]
```diff
@@ -128,33 +114,19 @@ inputs:
...
-        in the value of this input Xcode will simply use the Configuration
-        specified by the Scheme and will silently ignore this parameter!
+        (Optional) The name of the Xcode Configuration to use (Debug, Release, etc.). By default your Scheme's archive action defines which Configuration should be used, but this can be overridden it with this option.
+
+        **If the Configuration specified in this input does not exist in your project, the Step will silently ignore the value, and fall back to using the Configuration specified in the Scheme.**
   - disable_index_while_building: "yes"
     opts:
-      title: Disable indexing during the build
-      summary: Could make the build faster by disabling the indexing during the build run.
+      title: Disable Indexing During the Build
```
Comment (zoltan-baba) [inline step.yml:124]: Disable indexing during the build

File: step.yml [hunk 6]
```diff
@@ -163,61 +135,64 @@ inputs:
...
-        Only on rare occasions do you need to set the flag to "yes", usually when working with certain test cases or third-party dependencies.
+
+        On rare occasions, you may need to set the flag to "yes" — usually when working with certain test cases or third-party dependencies.
       value_options:
         - "yes"
         - "no"
   - cache_level: swift_packages
     opts:
-      title: Enable caching of Swift Package Manager packages
+      title: Enable Caching of Swift Package Manager Packages
```
Comment (zoltan-baba) [inline step.yml:145]: Enable caching of Swift Package Manager packages

File: step.yml [hunk 7]
```diff
@@ -163,61 +135,64 @@ inputs:
...
+        - `swift_packages`: Cache Swift Package Manager packages added to the Xcode project
       value_options:
       - "none"
       - "swift_packages"
       is_required: true
   - xcodebuild_options: ""
     opts:
       category: Debug
-      title: Additional options for xcodebuild call
+      title: Additional `xcodebuild` Options
```
Comment (zoltan-baba) [inline step.yml:158]: options

File: step.yml [hunk 8]
```diff
@@ -163,61 +135,64 @@ inputs:
...
-        This directory will contain the generated .app artifacts.
+      title: Output Directory Path
+      summary: The directory in which to place generated `.app` artifacts
+      description: The directory in which to place generated `.app` artifacts
       is_required: "true"
   - is_clean_build: "no"
     opts:
       category: Debug
-      title: Do a clean Xcode build before the build?
+      title: Do a Clean Xcode Build Before the Build?
```
Comment (zoltan-baba) [inline step.yml:179]: Do a clean Xcode build before the build?

File: step.yml [hunk 9]
```diff
@@ -226,8 +201,8 @@ inputs:
   - verbose_log: "no"
     opts:
       category: Debug
-      title: Enable verbose logging?
-      description: Enable verbose logging?
+      title: Verbose Logging
```
Comment (zoltan-baba) [inline step.yml:204]: logging

File: step.yml [hunk 10]
```diff
@@ -236,20 +211,19 @@ inputs:
 outputs:
   - BITRISE_APP_DIR_PATH:
     opts:
-      title: The generated (and copied) app directory
+      title: Generated (and Copied) App Directory
   - BITRISE_APP_DIR_PATH_LIST:
     opts:
-      title: List of the generated app paths
-      summary: "List of the generated (and copied) app paths"
+      title: List of the Generated App Paths
```
Comment (zoltan-baba) [inline step.yml:217]: generated app paths

File: step.yml [hunk 11]
```diff
@@ -236,20 +211,19 @@ inputs:
...
-        This output will include the main target app's path + every dependent target's app path.
-
+        This output will include the main target app's path, plus every dependent target's app path.
 
-        The paths are separated with `|` character, eg: `/deploy109787178/sample-apps-ios-workspace-swift.app|/deploy109787178/bitfall.sample-apps-ios-workspace-swift-watch.app`
+        The paths are separated by a `|` (pipe) character. (Example: `/deploy109787178/sample-apps-ios-workspace-swift.app|/deploy109787178/bitfall.sample-apps-ios-workspace-swift-watch.app`)
   - BITRISE_XCODE_BUILD_RAW_RESULT_TEXT_PATH:
     opts:
-      title: The full, raw build output file path
+      title: Full Raw Build Output File Path
```
Comment (zoltan-baba) [inline step.yml:225]:
```suggestion
      title: Full raw build output filepath
```

**Review by zoltan-baba — CHANGES_REQUESTED** [review]

**Comment by hisaac** [comment]
Thanks for all the good suggestions @zoltan-baba! I believe I've addressed them all in https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/35/commits/aab926a0ff96f00f7561305ceec68ad360bb0edb. Let me know if you see anything else.

**Review by zoltan-baba — DISMISSED** [review]

**Comment by hisaac** [comment]
@ofalvai I've updated the input title/summary/description info to match the conventions you sent me in https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/35/commits/6bf397c8663023c50732594cf07d368bac19df25 👍

**Review by ofalvai — APPROVED** [review]

**Pull request merged**